blueocean-core-js/src/less/header.less

```diff
@@ -128,3 +132,63 @@
         stroke: white;
     }
 }
+
+.jenkins-pipeline-header {
+    display: inline-block;
+}
+
+.TextInput.search-pipelines-input {
+    background-color: transparent;
+    background-repeat: no-repeat;
+    margin-left: 20px;
+
+    .TextInput-icon.u-icon-left {
+        pointer-events: none;
+        left: 0;
+
+        path {
+            fill: #ffffff;
+        }
+    }
+
+    input.TextInput-control {
+        border: none;
+        color: #ffffff;
+        font-size: 18px;
+        padding-top: 0px;
+        padding-bottom: 0px;
+        padding-left: 30px;
+        cursor: pointer;
+
+        &::-webkit-input-placeholder {
+            color: transparent;
+        }
+        &::-moz-placeholder {
+            color: transparent;
+        }
+        &:-ms-input-placeholder {
+            color: transparent;
+        }
+
+        &[value=""] {
+            width: 20px;
+            padding-left: 25px;
+        }
+
+        &:focus {
+            padding-left: 30px;
+            width: auto;
+            cursor: inherit;
+
+            &::-webkit-input-placeholder {
+                color: rgba(255, 255, 255, 0.4);
+            }
+            &::-moz-placeholder {
+                color: rgba(255, 255, 255, 0.4);
+            }
+            &:-ms-input-placeholder {
+                color: rgba(255, 255, 255, 0.4);
+            }
+        }
+    }
+}
```

Comment:
Please add a TODO here as a reminder that this code needs to get moved out.. and file a JIRA ticket to track the work.